assets/src/apps/authoring/components/RightMenu/RightMenu.tsx

```diff
@@ -88,6 +88,12 @@ const RightMenu: React.FC<any> = () => {
   const [questionBankData, setBankData] = useState<any>();
   const [questionBankSchema, setBankSchema] = useState<JSONSchema7>();
   const [showConfirmDelete, setShowConfirmDelete] = useState<boolean>(false);
+
+  const selectedPartDef = useMemo(
+    () => findPartByIdInActivity(currentActivity, currentPartSelection),
+    [currentActivity, currentPartSelection],
+  );
+
   useEffect(() => {
```

Comment:
There are a lot of useState / useEffect combos in RightMenu that could be refactored to useMemo to make them simpler to understand and reduce the number of render cycles the component needs